shared/js/client_api/channels.js

```diff
@@ -9,9 +9,9 @@ module.exports = {
       type: "get",
       dataType: "json",
     }).done(function(data, text) {
-      callback(data);
+      callback(null, data);
     }).fail(function(request, status, error) {
-      callback(null);
+      callback(error, null);
```

Comment:
I just discovered that with jquery, error here is not an error object but a string. Instead I want to do something like we have in apps client api:
```
callback(
          new Error('textStatus: ' + textStatus + '; errorThrown: ' + errorThrown),
          null);